janitor/io.py

```diff
@@ -132,91 +134,412 @@ def xlsx_table(
+        check("start_point", start_point, [str, int])
+        check("end_point", end_point, [str, int])
+
+    defaults = (
+        "value",
+        "internal_value",
+        "coordinate",
+        "row",
+        "column",
+        "data_type",
+        "is_date",
+        "number_format",
+    )
+
+    parameters = {
+        "fill": fill,
+        "font": font,
+        "alignment": alignment,
+        "border": border,
+        "protection": protection,
+        "comment": comment,
+    }
+
+    if kwargs:
+        if path_is_workbook:
+            if path.read_only:
+                _cell = ReadOnlyCell
+            else:
+                _cell = Cell
+        else:
+            if read_only:
+                _cell = ReadOnlyCell
+            else:
+                _cell = Cell
+
+        attrs = {
+            attr
+            for attr, _ in inspect.getmembers(_cell, not (inspect.isroutine))
+            if not attr.startswith("_")
+        }
+
+        for key in kwargs:
+            if key in defaults:
+                raise ValueError(
+                    f"{key} is part of the default attributes "
+                    "returned as a column."
+                )
+            elif key not in attrs:
+                raise AttributeError(
+                    f"{key} is not a recognized attribute of {_cell}."
+                )
+        parameters.update(kwargs)
+
+    if sheetnames:
+        check("sheetnames", sheetnames, [str, list, tuple])
+        if isinstance(sheetnames, str):
+            out = _xlsx_cells(
+                path,
+                sheetnames,
+                defaults,
+                parameters,
+                start_point,
+                end_point,
+                include_blank_cells,
+            )
+        else:
+            out = {
+                sheetname: _xlsx_cells(
+                    path,
+                    sheetname,
+                    defaults,
+                    parameters,
+                    start_point,
+                    end_point,
+                    include_blank_cells,
+                )
+                for sheetname in sheetnames
+            }
+    else:
+        out = {
+            sheetname: _xlsx_cells(
+                path,
+                sheetname,
+                defaults,
+                parameters,
+                start_point,
+                end_point,
+                include_blank_cells,
+            )
+            for sheetname in path.sheetnames
+        }
+
+    if (not path_is_workbook) and path.read_only:
+        path.close()
+
+    return out
+
+
+def _xlsx_cells(
+    wb,
+    sheetname: str,
+    defaults: tuple,
+    parameters: dict,
+    start_point: Union[str, int],
+    end_point: Union[str, int],
+    include_blank_cells: bool,
+):
+    """
+    Function to process a single sheet.
+    Returns a DataFrame.
+
+    :param wb: Openpyxl Workbook.
+    :param sheetname: Name of the sheet
+        from which the cells are to be extracted.
```

Comment:
How about just making the input here the worksheet itself, instead of workbook + sheetname? I don't see `wb` being used for any other purpose other than to access the worksheet.

Reply:
the user might have the workbook to access sheet names, hold on to a connection because he needs do more stuff; whereas within this function once we are done we close the connection

Reply:
So, to be clear, my previous comment I was referring to the parameters of the private function `_xlsx_cells` rather than the public one.

I understood your point about maybe the user already having a workbook on hand, but I don't quite get where this --- "whereas within this function once we are done we close the connection" --- happens.

Reply:
@thatlittleboy if U don't mind, kindly post a visual of what U mean

Reply:
made an edit to the code; when you can, lemme know if that is what you have in mind.

Reply:
@samukweku yep I think your changes are what I meant (having the parameters to `_xlsx_cells` as the worksheet itself rather than the full workbook + the sheetname.)
Thanks!